### DISCUSSION TITLE
How to do acknowledge re-used components

### DISCUSSION THREAD

**Discussion by rgieseke** [mhkeller/layercake#175] [General]
A question I had (also related to #169) is how to properly acknowledge re-used components from Layer Cake.
So far I have put a .md file in a directory `lib/components/layercake` with re-used/adapted components with something like

> Components in this folder originally from https://github.com/mhkeller/layercake under:
> 
> The MIT License (MIT)
> 
> Copyright (c) 2021 Michael Keller
> 
> Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the "Software"), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
> 
> The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
> 
> THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
> 

Do you have any thoughts on this @mhkeller ?
Should the components have a copyright line so this is clear?

**Comment by mhkeller** [comment]
That seems fine! I just updated the date in the License. The whole repo has an MIT license and so that pertains to the components, too.

**Comment by rgieseke** [comment]
That's good to know. But what if you copy only a single component, possibly with an approach discussed in #169 ?

It might make more sense to not have all Layer Cake components in a single directory, they could be organized by component in lib or go along with a page.

If I download a component directly, for example from https://layercake.graphics/components/AxisRadial.svelte, there is no indication about the license.

So I think it might be useful to include a link or license notice in the component file and maybe clarify in the docs how you would like that to be handled.
